common/src/main/java/com/wynntils/wynn/model/map/MapModel.java

```diff
@@ -33,31 +31,32 @@
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Optional;
 import java.util.Set;
-import java.util.concurrent.CompletableFuture;
 import java.util.stream.Stream;
 
 public final class MapModel extends Model {
     private static final String PLACES_JSON_URL =
             "https://raw.githubusercontent.com/Wynntils/Reference/main/locations/places.json";
     private static final String SERVICES_JSON_URL =
             "https://raw.githubusercontent.com/Wynntils/Reference/main/locations/services.json";
+    private static final String MAPS_JSON_URL =
+            "https://raw.githubusercontent.com/Wynntils/WynntilsWebsite-API/master/maps/maps.json";
+
     private static final Gson GSON = new GsonBuilder().create();
-    private static List<MapProfile> maps = new ArrayList<>();
+    private static List<MapTexture> maps = Collections.synchronizedList(new ArrayList());
```

Comment:
Theoretically this is a perfect use case for `CopyOnWriteArrayList`?

Reply:
"This is ordinarily too costly, but may be more efficient than alternatives when traversal operations vastly outnumber mutations, and is useful when you cannot or don't want to synchronize traversals, yet need to preclude interference among concurrent threads. "

I dunno. Maybe? If it makes you happy. :)

Reply:
In my mind, since the sync list is blocking and we render the minimap and the main map at the same time (if the main map screen is open), they would block each other in render. I am not sure about the impact though.

Reply:
Nah, they won't block, rendering is single threaded. The synchronization is just for when loading the maps. But I'll change it, won't hurt.